src/authModel.js

```diff
@@ -0,0 +1,89 @@
+const Token = require('./models/auth/Token')
+const Client = require('./models/auth/Client')
+const AuthorizationCode = require('./models/auth/AuthorizationCode')
+
+module.exports = {
+  getAccessToken: accessToken => {
+    console.log('finding token, accessToken = ' + accessToken)
+    return Token.findOne({ accessToken })
+      .populate('user')
+  },
+
+  getAuthorizationCode: authCode => {
+    console.log('finding auth code, code = ' + authCode)
+    return AuthorizationCode.findOne({ authorizationCode: authCode })
+      .populate('user client')
+  },
+
+  getClient: (clientId, clientSecret) => {
+    console.log('finding client, id = ' + clientId + ', secret = ' + clientSecret)
+    const params = { _id: clientId }
+    if (clientSecret) params.secret = clientSecret
+    return Client.findOne(params)
+  },
+
+  getRefreshToken: refreshToken =>
+    Token.find({ refreshToken })
+      .populate('user'),
+      // .exec(),
+
+  revokeAuthorizationCode: authCode => {
+    // should return true if successful
+    console.log('revoking auth code: ')
+    return AuthorizationCode.findOneAndRemove({ authorizationCode: authCode.authorizationCode })
+      .then(doc => !!doc)
```

Comment:
https://docs.mongodb.com/manual/reference/command/findAndModify/#findandmodify-value-field

If the document isn't found it will return value as null